mne/beamformer/_lcmv.py

```diff
@@ -24,6 +24,16 @@
 from ..channels.channels import _contains_ch_type
 
 
+def _deprecate_picks(data_obj, picks):
+    if picks is not None:
+        warn('Specifying picks is deprecated and will be removed in 0.16. '
+             'Specifying the selection of channels in info and setting picks '
+             'to None will remove this warning.',
+             DeprecationWarning)
+
+        data_obj.info = pick_info(data_obj.info, picks)
```

Comment:
This mismatch will be solved within the ``apply_`` functions by calling ``_pick_channels_spatial_filter`` - to make sure that the order of channels in the data is consistent with the spatial filter. Data (evoked, epochs, raw) is not needed within ``make_lcmv``, here, the consistency of forward model, data and noise covariance channel selections are checked and adjusted.